addons/mail/models/mail_thread.py

```diff
@@ -4145,6 +4145,9 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         empty_messages._cleanup_side_records()
         empty_messages.write({'pinned_at': None})
 
+        MessageTranslationSudo = self.env["mail.message.translation"].sudo()
+        MessageTranslationSudo.search([("message_id", "=", message.id)]).unlink()
+
         self.env['bus.bus']._sendone(message._bus_notification_target(), 'mail.record/insert', {
```

Comment:
bus notification should send the fact that the translation should be discarded for all tabs/clients

Reply:
Correct me if I am wrong but the only listener of message update in the chatter is the partner_id of the user that did the action. The other clients are not notified by the bus and have to reload their browser to notice the content of a chatter message has been modified. The reloading discard any potential translationValue anyway so there should not be any risk.

Note: I did not know it was possible to reach this method without modifying the body, I wrapped my code in a new if statement to handle that case.

Reply:
For now at least indeed, so you cannot test with another user. But it's already useful for the current user for other tabs/device, and it's a good idea to make the payload correct also for the day we're finally able to send update to all users viewing it.

Reply:
I left the payload unchanged as I believe comparing the content of the message before and after the update is sufficient.
https://github.com/odoo/odoo/pull/126555/files#diff-15ad441e03e5d430baa48e281c85675b0f700ad654f6d805f86d098003606267L101

Reply:
Client side knowledge is partial by design, the message might not yet be known or have an outdated value for any reason (in particular for chatter as it is not sync).

It's better generally speaking to send data rather than having heuristics to guess.
Also there is ongoing progress made by Alex to simplify insert payload to just update models, introducing extra processing on them goes against that.

Reply:
This information is now sent from the server via the bus using a new boolean value named "discard_translation" that is interpreted by the message_model itself.